src/pages/Questionnaire.tsx

```diff
@@ -1,20 +1,245 @@
+  width: 100%;
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 20px;
+
+  .progress-segment {
+    width: 100%;
+    height: 8px;
+    background-color: #ccc;
+    border-radius: 4px;
+    margin-right: 4px;
+
+    &.filled {
+      background-color: #07889b;
+    }
+  }
+`;
+
+const ProgressIndicator = styled.div`
+  font-size: 1.15rem;
+  color: #07889b;
+  font-weight: bold;
+  margin-bottom: 20px;
+  text-align: center;
+`;
+
+const NavigationButton = styled.button`
+  background-color: #07889b;
+  color: white;
+  border: none;
+  padding: 10px 20px;
+  border-radius: 20px;
+  cursor: pointer;
+  font-size: 1rem;
+  margin: 10px;
+  width: 150px;
+
+  &:hover {
+    background-color: #066a83;
+  }
+
+  &:disabled {
+    background-color: #ccc;
+    cursor: not-allowed;
+  }
+
+  @media (max-width: 768px) {
+    padding: 8px 16px;
+    width: 100px;
+    font-size: 0.9rem;
+  }
+`;
+
+const SubmitButton = styled(NavigationButton).attrs({ as: Link })`
+  display: inline-block;
+  text-align: center;
+  text-decoration: none;
+`;
 
 export function Questionnaire() {
-  const activeLink = links.find(
-    (link) => window.location.pathname === link.href
-  );
-  const activeTitle = activeLink ? activeLink.text : "Dashboard";
+  const [currentQuestion, setCurrentQuestion] = useState(0);
+
+  const questions: QuesProps[] = questionData.questions.map((question) => ({
+    ...question,
+    type: Type[question.type.toUpperCase() as keyof typeof Type],
+  }));
+
+  const handleNext = () => {
+    if (currentQuestion < questions.length - 1) {
+      setCurrentQuestion(currentQuestion + 1);
+    }
+  };
+
+  const handlePrevious = () => {
+    if (currentQuestion > 0) {
+      setCurrentQuestion(currentQuestion - 1);
+    }
+  };
+
+  const handleSubmit = () => {
+    console.log("Submit questionnaire");
+  };
 
   return (
-    <div className="full-page-bg">
-      <div className="dashboard-container">
-        <div className="dashboard-card-container">
-          <DashboardCard title={activeTitle} />
-        </div>
-      </div>
-    </div>
+    <PageBackground>
+      <DashboardContainer>
+        <DashboardCardContainer>
+          <StyledCard>
+            <QuesContainer>
+              <ProgressBar>
+                {questions.map((_, index) => (
+                  <div
+                    key={index}
+                    className={`progress-segment ${
+                      index <= currentQuestion ? "filled" : ""
+                    }`}
+                  ></div>
+                ))}
+              </ProgressBar>
+              <ProgressIndicator>
+                Question {currentQuestion + 1} / {questions.length}
+              </ProgressIndicator>
+              <Ques
+                id={questions[currentQuestion].id}
+                title={questions[currentQuestion].title}
+                type={questions[currentQuestion].type}
+                options={questions[currentQuestion].options}
```

Comment:
instead of passing each and every key, you can simply pass the whole: `questions[currentQuestion]` and then extract each key in the `Ques` component